runelite-client/src/main/java/net/runelite/client/plugins/microbot/nmz/NmzScript.java

```diff
@@ -150,27 +159,30 @@ public boolean interactWithObject(int objectId) {
         return false;
     }
 
-    public void manageLocatorOrb() {
-        if (Rs2Inventory.hasItem(ItemID.LOCATOR_ORB)) {
-            handleLocatorOrbUsage();
-            randomlyToggleRapidHeal();
-        } else if (Rs2Inventory.hasItem(ItemID.DWARVEN_ROCK_CAKE_7510)) {
-            handleDwarvenRockCake();
-            randomlyToggleRapidHeal();
-        }
-    }
+    public void manageSelfHarm() {
+        int currentHP = Microbot.getClient().getBoostedSkillLevel(Skill.HITPOINTS);
+        int currentRangedLevel = Microbot.getClient().getBoostedSkillLevel(Skill.RANGED);
+        int realRangedLevel = Microbot.getClient().getRealSkillLevel(Skill.RANGED);
+        boolean hasOverloadPotions = config.overloadPotionAmount() > 0;
+
+        if (currentHP >= maxHealth
+                && !useOverload
+                && (!hasOverloadPotions || currentRangedLevel != realRangedLevel)) {
+            maxHealth = 1;
+
+            if (Rs2Inventory.hasItem(ItemID.LOCATOR_ORB)) {
+                Rs2Inventory.interact(ItemID.LOCATOR_ORB, "feel");
+            } else if (Rs2Inventory.hasItem(ItemID.DWARVEN_ROCK_CAKE_7510)) {
+                Rs2Inventory.interact(ItemID.DWARVEN_ROCK_CAKE_7510, "guzzle");
+            }
 
-    private void handleDwarvenRockCake() {
-        if (Microbot.getClient().getBoostedSkillLevel(Skill.HITPOINTS) > Random.random(1, 5) && !useOverload
-                && Microbot.getClient().getBoostedSkillLevel(Skill.RANGED) != Microbot.getClient().getRealSkillLevel(Skill.RANGED)) {
-            Rs2Inventory.interact(ItemID.DWARVEN_ROCK_CAKE_7510, "guzzle");
+            if (currentHP == 1) {
+                maxHealth = Random.random(2, 8);
+            }
         }
-    }
 
-    public void handleLocatorOrbUsage() {
-        if (Microbot.getClient().getBoostedSkillLevel(Skill.HITPOINTS) > Random.random(1, 5) && !useOverload
-                && Microbot.getClient().getBoostedSkillLevel(Skill.RANGED) != Microbot.getClient().getRealSkillLevel(Skill.RANGED)) {
-            Rs2Inventory.interact(ItemID.LOCATOR_ORB, "feel");
+        if (config.randomlyTriggerRapidHeal()) {
```

Comment:
why is rapid heal applied here but not inside handleDwarvenRockCake?

It also seems that dwarenRockCake will set maxHealth

Reply:
I'm not that sure where rapid heal is normally used for within NMZ. So this part is a bit confusing to me.

Reply:
I've combined the two methods into this one because they were mostly doing the same thing. Only difference was the interaction with the item.